src/index.ts

```diff
@@ -22,16 +25,84 @@ import {
   TransferRequestParams,
 } from './types'
 
+const NETWORK_CHAIN_IDS = {
+  alfajores: 44787,
+  mainnet: 42220,
+}
+
+const fetch = fetchCookie(nodeFetch)
+
 export default class FiatConnectClient implements FiatConectApiClient {
   config: FiatConnectClientConfig
+  signingFunction: (message: string) => Promise<string>
+  _sessionExpiry?: Date
 
-  constructor(config: FiatConnectClientConfig) {
+  constructor(
+    config: FiatConnectClientConfig,
+    signingFunction: (message: string) => Promise<string>,
+  ) {
     this.config = config
+    this.signingFunction = signingFunction
+  }
+
+  async _ensureLogin() {
+    const loginResult = await this.login()
+    if (!loginResult.ok) {
+      throw new Error(`Login failed: ${loginResult.val.error}`)
+    }
+  }
+
+  /**
+   * Logs in with the provider and initializes a session.
+   *
+   * @returns a Promise resolving to the literal string 'success' on a
+   * successful login or an Error response.
+   */
+  async login(): Promise<Result<string, ErrorResponse>> {
+    try {
+      if (this._sessionExpiry && this._sessionExpiry < new Date()) {
+        return Ok('success')
+      }
+      const expirationDate = new Date(Date.now() + 14400000) // 4 hours from now
```

Comment:
is using the /clock endpoint in a separate issue? (I think so, but ZH isn't loading for me so hard to check rn)

Reply:
yes that's on a different issue: https://github.com/valora-inc/in-house-liquidity/issues/569